e2e/openshift/openshift_packaging_test.go

```diff
@@ -0,0 +1,268 @@
+  displayName: Cockroach Test Demo
+  image: {{DockerRegistry}}/cockroachdb-operator-index:{{AppVersion}}`
+
+var subscription = `apiVersion: operators.coreos.com/v1alpha1
+kind: Subscription
+metadata:
+  name: myoperator
+  namespace: test
+spec:
+  channel: stable
+  installPlanApproval: Automatic
+  name: cockroachdb-certified
+  source: cockroach-test
+  sourceNamespace: openshift-marketplace
+  startingCSV: cockroach-operator.{{AppVersion}}`
+
+type YAMLTemplate struct {
+	AppVersion     string
+	DockerRegistry string
+}
+
+var delete = []string{
+	"delete",
+	"-n",
+}
+
+var removeMap = map[string]string{
+	"operatorgroup": "test",
+	"subscription":  "myoperator",
+}
+
+func TestPackaging(t *testing.T) {
+
+	appVersion := os.Getenv("APP_VERSION")
+	require.True(t, appVersion != "", "APP_VERSION env var not set")
+	dockerRegistry := os.Getenv("DOCKER_REGISTRY")
+	require.True(t, dockerRegistry != "", "DOCKER_REGISTRY env var not set")
+	kubeconfig := os.Getenv("KUBECONFIG")
+	require.True(t, kubeconfig != "", "KUBECONFIG evn var not set")
+
+	args := []string{
+		"delete",
+		"crd",
+		"crdbclusters.crdb.cockroachlabs.com",
+	}
+	process.ExecJUnit("oc", args, os.Environ())
+
+	catalog := append(delete, "openshift-marketplace", "CatalogSource", "cockroach-test")
+	process.ExecJUnit("oc", catalog, os.Environ())
+
+	for k, v := range removeMap {
+		d := append(delete, "test", k, v)
+		process.ExecJUnit("oc", d, os.Environ())
+	}
+
+	args = []string{
+		"delete",
+		"namespace",
+		"test",
+	}
+
+	process.ExecJUnit("oc", args, os.Environ())
+
+	args = []string{
+		"new-project",
+		"test",
+	}
+
+	require.NoError(t, process.ExecJUnit("oc", args, os.Environ()), "creating new project with oc failed")
+
+	yaml := &YAMLTemplate{
+		AppVersion:     appVersion,
+		DockerRegistry: dockerRegistry,
+	}
+
+	catalogByte := parseTmpl(t, catalogSource, yaml)
+	subByte := parseTmpl(t, subscription, yaml)
+
+	catalogFilename := writeTempFile(t, catalogByte)
+	subFilename := writeTempFile(t, subByte)
+
+	var b bytes.Buffer
+	b.WriteString(operatorGroup)
+
+	operatorFilename := writeTempFile(t, b.Bytes())
+
+	args = []string{
+		"create", "-f",
+		catalogFilename,
+		subFilename,
+		operatorFilename,
+	}
+
+	require.NoError(t, process.ExecJUnit("oc", args, os.Environ()), "failed creating openshift operator files")
+	cs, err := createClientset()
+	require.NoError(t, err, "failed to create clientset")
+	require.NoError(t, testPods(t, cs), "failed finding operator")
+
+	args = []string{
+		"create", "-f", "examples/example.yaml",
+	}
+
+	require.NoError(t, process.ExecJUnit("oc", args, os.Environ()), "failed creating crdb database")
+
+	testLog := zapr.NewLogger(zaptest.NewLogger(t))
+
+	require.NoError(t,
+		kube.WaitUntilAllStsPodsAreReady(
+			context.TODO(),
+			cs,
+			testLog,
+			"crdb",
+			"test",
+			800*time.Second,
+			10*time.Second,
+		),
+		"crdb database did not start",
+	)
+}
+
+func writeTempFile(t *testing.T, b []byte) string {
```

Comment:
I cannot use temp files because they are not available to the subprocess when I shell out.